server/NetPacksServer.cpp

```diff
@@ -272,7 +272,9 @@ bool HireHero::applyGh(CGameHandler * gh)
 
 bool BuildBoat::applyGh(CGameHandler * gh)
 {
-	throwOnWrongOwner(gh, objid);
+	if(gh->getPlayerRelations(gh->getOwner(objid), gh->getPlayerAt(c)) == PlayerRelations::ENEMIES)
+		throwAndCompain(gh, "Can't build boat at enemy shipyard");

```

Comment:
Isn't this a typo?

Reply:
Sure it is. Wanna me to fix it?

Reply:
I also think I can extract a method throwWnenOwnerIsEnemy and use it for recruit hero

Reply:
I also think I can extract a method throwWnenOwnerIsEnemy and use it for recruit hero